hopp/simulation/technologies/generic/generic_multi.py

```diff
@@ -0,0 +1,248 @@
+    capacity_factor: float = field(init = False)
+    annual_energy_pre_curtailment_ac: float = field(init = False)
+    
+    def __attrs_post_init__(self):
+        """Initialize some attributes and set defaults as needed. This method does the following:
+        
+        1) ensures that subsystem_names are unique and reassigns names to each subsystem if needed
+
+        2) updates generation profile and system capacity. This initializes the GenericMultiSystem attributes:
+            gen, annual_energy, annual_energy_pre_curtailment_ac, system_capacity, system_capacity_ac, and capacity_factor.
+        """
+        if len(self.subsystem_names)==0:
+            subsystem_names_original = [sub.system_name for sub in self.subsystem_names]
+            for ni,sub_name in enumerate(subsystem_names_original):
+                if subsystem_names_original.count(sub_name)>1:
+                    subsystem_names_original[ni] = f"{sub_name}_{ni}"
+                    self.subsystems[ni].value("system_name", f"{sub_name}_{ni}")
+            self.subsystem_names = subsystem_names_original
+        self.system_capacity = 0.0 #temporarily set to avoid attribute error when calculating capacity factor
+        self.update_generation_profile(None)
+        self.update_system_capacity(None)
+
+    def value(self, name:str, set_value=None):
+        """Set or retrieve attribute of `hopp.simulation.technologies.generic.generic_plant.GenericSystem`.
+            if set_value = None, then retrieve value; otherwise overwrite variable's value.
+        
+        Args:
+            name (str): name of attribute to set or retrieve.
+            set_value (Optional): value to set for variable `name`. 
+                If `None`, then retrieve value. Defaults to None.
+        """
+
+        if set_value is not None:
+            self.__setattr__(name, set_value)
+        else:
+            return self.__getattribute__(name)
+    
+    def execute(self, project_life):
+        """Empty execute function since generation is set during initialization.
+
+        Args:
+            project_life (int): unused project life in years
+        """
+        return
+
+    def export(self):
+        """Return all the generic system configuration in a dictionary for the financial model
+        
+        Returns:
+            dict: generic system configuration for the financial model.
+        """
+
+        config = {
+            'system_capacity': self.system_capacity,
+        }
+        return config
+
+    def update_generation_profile(self, placeholder):
+        """Recalculate ``gen`` attribute after subsystem generation profiles may have been updated. Also updates ``annual_energy``, 
+        ``annual_energy_pre_curtailment_ac``, and ``capacity_factor``.
+
+        Args:
+            placeholder (None): unused placeholder value so this function parallels 
+            the function in ``hopp.simulation.technologies.generic.generic_plant.GenericSystem``
+        """
```

Comment:
I don't love the use of `placeholder` here, though I get your logic of wanting to parallel the structure of the other method. Is there a way to do this well without that placeholder? Looking at this a bit, it seems like you're usually passing `None` into here, so it's not usually even needed? Maybe I'm misunderstanding.